### PR TITLE
crypto: benchmark BLS single signing and verification

### PR DESCRIPTION
**Pull request by AdoAdoAdo**
go test crypto/signing/mcl/singlesig/bls_bench_test.go -bench BlsSingleSigner

i7 7th gen 
BenchmarkBlsSingleSigner_Sign-4             3208            401445 ns/op
BenchmarkBlsSingleSigner_Verify-4            870           1628854 ns/op

### CODE REVIEW

**Comment by LucianMincu** [comment]
Arm64 Rasp Pi 4

```
go test crypto/signing/mcl/singlesig/bls_bench_test.go -bench BlsSingleSigner
goos: linux
goarch: arm64
BenchmarkBlsSingleSigner_Sign-4     	     704	   1711463 ns/op
BenchmarkBlsSingleSigner_Verify-4   	     140	   8492564 ns/op
PASS
ok  	command-line-arguments	54.556s
```

File: crypto/signing/mcl/singlesig/bls_bench_test.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+func BenchmarkBlsSingleSigner_Sign(b *testing.B) {
+	signer := singlesig.NewBlsSigner()
+	suite := mcl.NewSuiteBLS12()
+	kg := signing.NewKeyGenerator(suite)
+	privKey, _ := kg.GeneratePair()
+
+	var err error
+	nbMessages := 10000
+	messages := make([][]byte, 0, 10000)
```
Comment (andreibancioiu) [inline crypto/signing/mcl/singlesig/bls_bench_test.go:22]: Could have used `nbMessages`.

File: crypto/signing/mcl/singlesig/bls_bench_test.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+	hasher := sha256.Sha256{}
+
+	for i := 0; i < nbMessages; i++ {
+		strIdx := strconv.Itoa(i)
+		messages = append(messages, hasher.Compute(strIdx))
+	}
+
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		_, err = signer.Sign(privKey, messages[i%nbMessages])
```
Comment (andreibancioiu) [inline crypto/signing/mcl/singlesig/bls_bench_test.go:32]: Messages to sign are quite short. Maybe relevant to the benchmark?

File: crypto/signing/mcl/singlesig/bls_bench_test.go [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+func BenchmarkBlsSingleSigner_Verify(b *testing.B) {
+	signer := singlesig.NewBlsSigner()
+	suite := mcl.NewSuiteBLS12()
+	kg := signing.NewKeyGenerator(suite)
+	privKey, pubKey := kg.GeneratePair()
+
+	var err error
+	nbMessages := 10000
+	messages := make([][]byte, 0, 10000)
```
Comment (andreibancioiu) [inline crypto/signing/mcl/singlesig/bls_bench_test.go:45]: Could have used `nbMessages`.

File: crypto/signing/mcl/singlesig/bls_bench_test.go [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+		strIdx := strconv.Itoa(i)
+		messages = append(messages, hasher.Compute(strIdx))
+		signature, err := signer.Sign(privKey, messages[i%nbMessages])
+		require.Nil(b, err)
+		signatures = append(signatures, signature)
+	}
+
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		err = signer.Verify(pubKey, messages[i%nbMessages], signatures[i%nbMessages])
```
Comment (andreibancioiu) [inline crypto/signing/mcl/singlesig/bls_bench_test.go:59]: Messages to verify are a bit short (see above).

**Pull request merged**